### PR TITLE
Pv2 1643 use commitments data directly for migration

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

File: src/SFA.DAS.Payments.Migration/Program.cs [hunk 1]
```diff
@@ -560,15 +565,35 @@ private static string UpdateTableAndSchemaNames(string originalSql)
...
+                        commitment.ProgrammeType = int.Parse(portions[1]);
+                        commitment.PathwayCode = int.Parse(portions[2]);
+                    }
+                    else
+                    {
+                        await Log($"Unknown TrainingType: {commitment.TrainingType} for CommitmentId: {commitment.CommitmentId}");
+                    }
+                }
 
                 var commitmentsById = commitments.GroupBy(x => x.CommitmentId);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.Migration/Program.cs]: From looking at the code in approvals that populates the ApprenticeshipCreated event it looks like the grouping should be on the ApprenticeshipId not the commitmentid.  May need to confirm this with the approvals team.

File: src/SFA.DAS.Payments.Migration/Constants/DasSql.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                TrainingType, TrainingCode,  
+	            ApprenticeshipEmployerTypeOnApproval [ApprenticeshipEmployerType], 
+	            AccountLegalEntityPublicHashedId
+	            
+            FROM [dbo].[Apprenticeship] A
+            JOIN PriceHistory H
+	            ON A.Id = H.ApprenticeshipId
+            JOIN Commitment C
+	            ON C.Id = A.CommitmentId
+            WHERE A.PaymentStatus > 0
```
Comment (abroodie) [inline src/SFA.DAS.Payments.Migration/Constants/DasSql.cs]: the Apprenticeship table has a field called "IsApproved", might be worth using that instead.

File: src/SFA.DAS.Payments.Migration/Constants/DasSql.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿namespace SFA.DAS.Payments.Migration.Constants
+{
+    static class DasSql
+    {
+        public static string Commitments = @"
+            SELECT  A.Id [CommitmentId], A.Uln, ProviderId [Ukprn], EmployerAccountId [AccountId],
```
Comment (abroodie) [inline src/SFA.DAS.Payments.Migration/Constants/DasSql.cs]: Query looks very messy, can you tidy it up, e.g.  each column on a separate line.

File: src/SFA.DAS.Payments.Migration/Program.cs (hunk 1 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Migration/Program.cs]: The apprenticeship id is mapped to the property commitment id in the SQL to minimise the changes required for this ticket. This code hasn't changed

File: src/SFA.DAS.Payments.Migration/Constants/DasSql.cs (hunk 2 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Migration/Constants/DasSql.cs]: I had a chat with Chris and he recommended this route over `IsApproved`

File: src/SFA.DAS.Payments.Migration/Constants/DasSql.cs (hunk 3 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Migration/Constants/DasSql.cs]: Updated

File: src/SFA.DAS.Payments.Migration/Constants/DasSql.cs (hunk 2 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.Migration/Constants/DasSql.cs]: ok.

File: src/SFA.DAS.Payments.Migration/Program.cs (hunk 1 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.Migration/Program.cs]: looks ok now

File: src/SFA.DAS.Payments.Migration/Program.cs [hunk 4]
```diff
@@ -32,81 +32,48 @@ static async Task Log(string message)
...
 
-                if (typeinputAsInteger == 7)
-                {
-                    await ProcessFailedV1Payments();
-                }
-
-                await Log("Finished - press enter to continue...");
-                Console.ReadLine();
+                await MakeSelection(typeinputAsInteger);
+                goto START;
```
Comment (benagain) [inline src/SFA.DAS.Payments.Migration/Program.cs]: 😱

**Pull request merged**